addons/l10n_pe_website_sale/views/templates.xml

```diff
@@ -0,0 +1,115 @@
+                    type='hidden'/>
+            </t>
+        </div>
+
+    </template>
+
+    <template id="partner_address_info" name="Peruvian partner address">
+
+        <!-- show city -->
+        <div
+            t-attf-class="mb-3 #{error.get('city_id') and 'o_has_error' or ''} col-lg-6 div_city_id"
+            t-attf-style="#{(country and country.code != 'PE') and 'display: none' or ''}">
+            <label class="col-form-label" for="city_id">City</label>
+            <select
+                id="city_id"
+                name="city_id"
+                t-attf-class="form-select #{error.get('city_id') and 'is-invalid' or ''}"
+                data-init="1">
+                <t t-foreach="state_cities" t-as="city">
+                    <option
+                        t-att-value="city.id"
+                        t-att-selected="city.id == ('city_id' in checkout and checkout['city_id'] != '' and int(checkout['city_id']))">
+                        <t t-out="city.name" />
+                    </option>
+                </t>
+            </select>
+        </div>
+
+        <!-- show district -->
+        <div
+            t-attf-class="mb-3 #{error.get('l10n_pe_district') and 'o_has_error' or ''} col-lg-6 div_district"
+            t-att-style="not city and 'display: none'">
+            <label class="col-form-label" for="l10n_pe_district">District</label>
+            <select
+                id="l10n_pe_district"
+                name="l10n_pe_district"
+                t-attf-class="form-select #{error.get('l10n_pe_district') and 'is-invalid' or ''}"
+                data-init="1">
+                <t t-foreach="city_districts" t-as="district">
+                    <option
+                        t-att-value="district.id"
+                        t-att-selected="district.id == ('l10n_pe_district' in checkout and checkout['l10n_pe_district'] != '' and int(checkout['l10n_pe_district']))">
+                        <t t-out="district.name" />
+                    </option>
+                </t>
+            </select>
+        </div>
+
+    </template>
+
+    <template id="address" inherit_id="website_sale.address">
+        <xpath expr="//input[@name='vat']/.." position="before">
+            <t t-if="mode[1] == 'billing'" position="inside">
+                <t t-if="res_company.country_id.code == 'PE'">
+                    <t t-set="partner" t-value="website_sale_order.partner_id" />
+                    <t t-call="l10n_pe_website_sale.partner_info" />
+                </t>
+            </t>
+        </xpath>
+        <label for="vat" position="attributes">
+            <attribute name="t-if">res_company.country_id.code != 'PE'</attribute>
+        </label>
+        <label for="vat" position="after">
+            <label t-if="res_company.country_id.code == 'PE'" class="col-form-label label-optional" for="vat">
+                Number
+            </label>
+        </label>
```

Comment:
pass the company country code here, to avoid doing the rpc (it will be less code)
```suggestion
        </label>
        <xpath expr="//select[@name='country_id']" position="attributes">
            <attribute name="t-att-data-company_country_code">res_company.country_id.code</attribute>
        </xpath>
```

Reply:
Hello @JulienVR turns out I ended up taking care of these changes when I got back to work.

I added this the way you suggested but when I tried to obtain this value it wouldn't be available in other parts of /shop/something due to the widget having a generic selector for everything shop related. What I did was pass the value on the [layout template](https://github.com/odoo/odoo/blob/70a3e9e44e416d02b27f5c495b97593938dfd31f/addons/l10n_pe_website_sale/views/templates.xml#L112), this way, no matter where in /shop you are it will always be available.

Having said that, [this](https://github.com/odoo/odoo/blob/70a3e9e44e416d02b27f5c495b97593938dfd31f/addons/l10n_pe_website_sale/static/src/js/website_sale.js#L15) is what I did to retrieve the value since it no longer calls an rpc to do so.

Now, the [following conditionals](https://github.com/odoo/odoo/blob/70a3e9e44e416d02b27f5c495b97593938dfd31f/addons/l10n_pe_website_sale/static/src/js/website_sale.js#L16) are in case users want to add a different shipping address, applying all your changes were causing an error, something we probably did not consider before when testing out the code.